lib/web/integrations_awsoidc.go

```diff
@@ -1360,14 +1357,21 @@ func (h *Handler) awsOIDCConfigureIdP(w http.ResponseWriter, r *http.Request, p
 		return nil, trace.BadParameter("invalid role %q", role)
 	}
 
+	proxyAddr, err := oidc.IssuerFromPublicAddress(h.cfg.PublicProxyAddr, "")
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
 	// The script must execute the following command:
 	// teleport integration configure awsoidc-idp
 	argsList := []string{
 		"integration", "configure", "awsoidc-idp",
 		fmt.Sprintf("--cluster=%s", shsprintf.EscapeDefaultContext(clusterName)),
 		fmt.Sprintf("--name=%s", shsprintf.EscapeDefaultContext(integrationName)),
 		fmt.Sprintf("--role=%s", shsprintf.EscapeDefaultContext(role)),
+		fmt.Sprintf("--proxy-public-url=%s", shsprintf.EscapeDefaultContext(proxyAddr)),
```

Comment:
What is this for? The PR description makes it seem like it's not strictly related to dead code removal. ;)

Reply:
Instead of doing a `argsList = append(argsList, "--proxy-public-url=%s")`, I'm now adding it to the initial slice
We were appending here (note that `s3Bucket` and `s3Prefix` were always empty).
```go
	case s3Bucket == "" && s3Prefix == "":
		proxyAddr, err := oidc.IssuerFromPublicAddress(h.cfg.PublicProxyAddr, "")
		if err != nil {
			return nil, trace.Wrap(err)
		}

		argsList = append(argsList,
			fmt.Sprintf("--proxy-public-url=%s", shsprintf.EscapeDefaultContext(proxyAddr)),
		)
```